Slim/Formats/Playlists/CUE.pm

```diff
@@ -60,144 +122,578 @@ sub parse {
+					value		=> $value,
+					filename	=> $filename,
+					returned	=> $1
+				}));
+
+			} elsif ($inAlbum and $value =~ /^\"?(\S+)\"?/i) {
+
+				# Some cue sheets may not have quotes. Allow that, but
+				# the filenames can't have any spaces in them."
+				$filename = $embedded || $1;
+				$filename = Slim::Utils::Misc::fixPath($filename, $baseDir);
+
+				$filesSeen++;
+				
+				main::DEBUGLOG && $log->is_debug && $log->debug(Data::Dump::dump({
+					message		=> 'filename with no quotes',
+					line		=> $line,
+					command		=> $command,
+					value		=> $value,
+					filename	=> $filename,
+					returned	=> $1
+				}));
+
+			} elsif ($inAlbum){
+
+				# Invalid filename, skipped.
+				main::DEBUGLOG && $log->is_debug && $log->debug(Data::Dump::dump({
+					message		=> 'Invalid filename',
+					line		=> $line,
+					command		=> $command,
+					value		=> $value
+				}));
+
+			} elsif (defined $currtrack and defined $filename){
+
+				# Better remove this point.
+				# To me it doesn't do what its was meant to do.
+				#
+				# In any case it will not take effect due to 
+				# the patch below marked : 
+				#
+				# Bug 5735, skip cue sheets with multiple FILE entries.
+				# 
+				# Here original comments:
+				#
+				# Each track in a cue sheet can have a different
+				# filename. See Bug 2126 &
+				# http://www.hydrogenaudio.org/forums/index.php?act=ST&f=20&t=4586
 			
-		} elsif (defined $currtrack and
+				$tracks->{$currtrack}->{'FILENAME'} = $filename;
+			} 
+			
+			# TODO: Correctly Handle Multiple file cue sheet.
+			# http://www.hydrogenaudio.org/forums/index.php?act=ST&f=20&t=4586
 
-			$line =~ /^(?:\s+REM )?\s*(TITLE|YEAR|GENRE|COMMENT|COMPOSER|CONDUCTOR|BAND)\s+\"(.*)\"/i) {
+		} else {
 
-			$tracks->{$currtrack}->{uc $1} = $2;
+			# handle remaning Commands as a list of keys and values.
+			($cuesheet, $tracks)=_addCommand($cuesheet, 
+											 $tracks,
+											 $inAlbum,
+											 $currtrack,
+											 $command,
+											 _removeQuotes($value));
+		}
+	}
+	main::DEBUGLOG && $log->is_debug && $log->debug(Data::Dump::dump({
+		message		=> 'after line parsing',
+		cuesheet	=> $cuesheet,
+		tracks		=> $tracks,
+		filename	=> $filename
+	}));
 
-		} elsif (defined $currtrack and
+	# Bug 5735, skip cue sheets with multiple FILE entries
+	if ( $filesSeen > 1 ) {
+		$log->warn('Skipping cuesheet with multiple FILE entries');
+		return;
+	}
 
-			$line =~ /^(?:\s+REM )?\s*(?:DISC|DISCNUMBER)\s+\"?([^\"]+)\"?/i) {
+	# Here controls on the entyre cuesheet structure, moving attributes
+	# to the correct level, preventing duplicates and renaming when needed.
+	#
+	if (defined $cuesheet->{'TITLE'}){
+		$cuesheet->{'ALBUM'} = $cuesheet->{'TITLE'};
+		delete $cuesheet->{'TITLE'};
+	}
 
-			# Some Cue sheets may not bother with quotes around a number
-			$tracks->{$currtrack}->{'DISC'} = $1;
+	if (defined $cuesheet->{'PERFORMER'}){
 
-		} elsif (defined $currtrack and
+		$cuesheet->{'ARTIST'} = $cuesheet->{'PERFORMER'};
```

Comment:
Shouldn't these assignments be optional? Eg. what if there was an ALBUMARTIST _and_ a PERFORMER value? The former would be overridden. I'm not sure this is what we want.